### PR TITLE
add recommended program structure

### PR DESCRIPTION
**Pull request by paul-schaaf**

### CODE REVIEW

**Comment by paul-schaaf** [comment]
draft cause the book also needs to be adjusted after review

File: programs/tic-tac-toe/programs/tic-tac-toe/src/lib.rs [hunk 1]
```diff
@@ -1,9 +1,12 @@
...
-use num_traits::*;
-use std::mem;
-
 use anchor_lang::prelude::*;
 
+pub mod errors;
+pub mod instructions;
+pub mod state;
+
+use instructions::*;
```
Comment (armaniferrante) [inline programs/tic-tac-toe/programs/tic-tac-toe/src/lib.rs]: move this above with the other imports?

File: programs/tic-tac-toe/programs/tic-tac-toe/src/instructions/setup_game.rs [hunk 2]
```diff
@@ -0,0 +1,18 @@
+use anchor_lang::prelude::*;
+
```
Comment (armaniferrante) [inline programs/tic-tac-toe/programs/tic-tac-toe/src/instructions/setup_game.rs]: should remove this newline

Comment (armaniferrante) [reply programs/tic-tac-toe/programs/tic-tac-toe/src/instructions/setup_game.rs]: also in other files above

**Pull request merged**